python/cudf/cudf/tests/test_reshape.py

```diff
@@ -114,3 +118,89 @@ def test_df_stack(nulls, num_cols, num_rows, dtype):
 
     assert_eq(expect, got)
     pass
+
+@pytest.mark.parametrize("num_rows", [1, 2, 1000])
+@pytest.mark.parametrize("num_cols", [1, 2, 10])
+@pytest.mark.parametrize(
+    "dtype",
+    [
+        "int8",
+        "int16",
+        "int32",
+        "int64",
+        "float32",
+        "float64",
+        "datetime64[ms]",
+        "str",
+    ],
```

Comment:
I do not fully understand why this needs category tests and am having trouble determining how to make that work as a result.

Reply:
It would be good if we could support interleaving categorical columns but I understand the challenge. Instead we should probably throw an exception if someone tries to use a "category" dtype column with the API for now.